### PR TITLE
[Hist][Docs] Revise TH1 documentation.

### PR DESCRIPTION
**Pull request by hageboeck**
After being notified that beginners were unable to label the axes of a
histogram, I revised the TH1 documentation a bit, such that the words
"axis labels" can now be found on the doxygen page.
Some lvl4 headers were converted in lvl2, so there is a bit more
structure, and a simple table of contents was added. Unfortunately,
doxygen doesn't support an automatic table of contents in class
documentation.

### CODE REVIEW

File: hist/hist/src/TH1.cxx [hunk 1]
```diff
@@ -150,17 +159,46 @@ Histograms may also be created by:
...
+ the list of objects in memory.
+ When a file is closed, all histograms in memory associated with this file
+ are automatically deleted.
+
+### Labelling axes
+
+ Axis titles can be specified in the title argument of the constructor.
+ They must be separated by ";":
+~~~ {.cpp}
+        TH1F* h=new TH1F("h", "Histogram title;X Axis;Y Axis;Z Axis", 100, 0, 1);
```
Comment (vepadulano) [inline hist/hist/src/TH1.cxx:179]:
```suggestion
        TH1F* h=new TH1F("h", "Histogram title;X Axis;Y Axis", 100, 0, 1);
```

File: hist/hist/src/TH1.cxx [hunk 2]
```diff
@@ -150,17 +159,46 @@ Histograms may also be created by:
...
 ~~~
-     When the histogram is deleted, the reference to it is removed from
-     the list of objects in memory.
-     When a file is closed, all histograms in memory associated with this file
-     are automatically deleted.
+ The histogram title and the axis titles can be any TLatex string, and
+ are persist if a histogram is written to a file.
```
Comment (vepadulano) [inline hist/hist/src/TH1.cxx:182]:
```suggestion
 are persisted if a histogram is written to a file.
```

File: hist/hist/src/TH1.cxx (hunk 1 above)
Comment (vepadulano) [reply hist/hist/src/TH1.cxx:179]: Even though there is a Z axis even in TH1 I wouldn't show it as an example.

**Review by vepadulano — APPROVED** [review]
Nice! Minor comments

File: hist/hist/src/TH1.cxx (hunk 1 above)
Comment (hageboeck) [reply hist/hist/src/TH1.cxx]: Very valid point. Thanks! 👍

**Pull request merged**